spec/ruby/optional/capi/ext/util_spec.c

```diff
@@ -16,7 +16,14 @@ VALUE util_spec_rb_scan_args(VALUE self, VALUE argv, VALUE fmt, VALUE expected,
     args[i] = rb_ary_entry(argv, i);
   }
 
-  result = rb_scan_args(argc, args, RSTRING_PTR(fmt), &a1, &a2, &a3, &a4, &a5, &a6);
+  if (*RSTRING_PTR(fmt) == 'k') {
+#ifdef RB_SCAN_ARGS_KEYWORDS
+    result = rb_scan_args_kw(RB_SCAN_ARGS_KEYWORDS, argc, args, RSTRING_PTR(fmt)+1, &a1, &a2, &a3, &a4, &a5, &a6);
```

Comment:
Would `rb_scan_args_kw` work too if this branch was used for all specs on 2.7?
That way we would not need to pass this `k` prefix.

Reply:
We could use `rb_scan_args_kw` with `RB_SCAN_ARGS_LAST_HASH_KEYWORDS` and avoid the `'k'` prefix. However, I think the current approach of only calling `rb_scan_args_kw` when it is necessary is better. Mostly because `rb_scan_args` has a separate optimized implementation in `include/ruby/ruby.h` that would not otherwise be tested by these specs.

Reply:
OK, let's do that then :)